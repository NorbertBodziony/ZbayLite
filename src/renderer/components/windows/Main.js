import React from 'react'
import PropTypes from 'prop-types'
import { Route } from 'react-router-dom'
import * as R from 'ramda'

import Grid from '@material-ui/core/Grid'
import { withStyles } from '@material-ui/core/styles'

import WindowWrapper from '../ui/WindowWrapper'
import Sidebar from '../widgets/sidebar/Sidebar'
import Channel from '../../containers/pages/Channel'
import DirectMessages from '../../containers/pages/DirectMessages'
import DepositMoneyModal from '../../containers/ui/DepositMoneyModal'
import InvitationModal from '../../containers/ui/InvitationModal/InvitationModal'

const styles = {
  gridRoot: {
    'min-height': '100vh',
    'min-width': '100vw'
  }
}

export const Main = ({ match, classes }) => {
  return (
    <>
      <DepositMoneyModal />
      <InvitationModal />
      <WindowWrapper>
        <Grid container direction='row' className={classes.gridRoot}>
          <Grid item>
            <Sidebar />
          </Grid>
          <Grid item xs>
            <Route path={`${match.url}/channel/:id`} component={Channel} />
            <Route
              path={`${match.url}/direct-messages/:id/:username`}
              component={DirectMessages}
            />
          </Grid>
        </Grid>
      </WindowWrapper>
    </>
  )
}

Main.propTypes = {
  classes: PropTypes.object.isRequired,
  match: PropTypes.shape({
    url: PropTypes.string.isRequired
  }).isRequired
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(Main)
