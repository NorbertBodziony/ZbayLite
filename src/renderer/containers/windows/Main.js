import React from 'react'
import { Redirect } from 'react-router'
import { connect } from 'react-redux'

import MainComponent from '../../components/windows/Main'
import vaultSelectors from '../../store/selectors/vault'

export const mapStateToProps = state => ({
  vaultLocked: vaultSelectors.locked(state)
})

export const Main = ({ vaultLocked, ...props }) => (
  vaultLocked
    ? <Redirect to='/vault' />
    : <MainComponent {...props} />
)

export default connect(mapStateToProps)(Main)
