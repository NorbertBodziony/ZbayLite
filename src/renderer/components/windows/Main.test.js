/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'

import { Main } from './Main'
import { mockClasses } from '../../../shared/testing/mocks'

describe('Main', () => {
  it('renders component', () => {
    const result = shallow(<Main classes={mockClasses} match={{ url: 'test' }} />)
    expect(result).toMatchSnapshot()
  })
})
