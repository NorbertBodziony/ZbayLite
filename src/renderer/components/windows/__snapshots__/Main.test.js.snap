// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders component 1`] = `
<Memo(WithStyles(WindowWrapper))>
  <WithStyles(Grid)
    className="gridRoot-mock"
    container={true}
    direction="row"
  >
    <WithStyles(Grid)
      item={true}
    >
      <WithStyles(Sidebar) />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={true}
    >
      <Route
        component={[Function]}
        exact={true}
        path="test"
      />
      <Route
        component={[Function]}
        exact={true}
        path="test/channel/:id"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</Memo(WithStyles(WindowWrapper))>
`;
